pkg/lua/Makefile.lua

```diff
@@ -1,9 +1,9 @@
-SRC := $(filter-out lua.c luac.c,$(wildcard *.c))
+SRC := $(filter-out loadlib.c lua.c luac.c,$(wildcard *.c))
 
-# This builds for native using POSIX system calls and some extra libraries, and
-# removes a compiler warning that warns against using tmpnam().
-ifeq ($(BOARD),native)
-  CFLAGS += -DLUA_USE_LINUX
-endif
+CFLAGS += -fstack-usage -fconserve-stack \
+          -DLUA_MAXCAPTURES=16 \
+          -DL_MAXLENNUM=50
+#    Enable these options to debug stack usage
+#          -Wstack-usage=128 -Wno-error=stack-usage=128
 
```

Comment:
Note that now the examples build with warnings...
@jcarrano @jia200x don't want to do a blocking comment but wanted to highlight this

Reply:
If you are referring to this:
```
 loslib.c:(.text.os_tmpname+0x29): warning: the use of `tmpnam' is dangerous, better use `mkstemp'
```
There is no standard C way of fixing it, without changing the OS module's API. Asking for a filename and then creating the file is a bad idea, but that's how the lua standard library is made to work.

Reply:
@jcarrano is it possible to patch that file?

Reply:
Is ` CFLAGS += -DLUA_USE_LINUX` a bad idea

Reply:
It is already patched. Here is what happens:
1. The `os` module contains a function `os.tempname` which should return a filename suitable for a temporary file.
2. tmpnam is a C function that was supposed to do this before peopl realized that it's not good to generate a filename without creating the file at the same time: it exposes you to a race condition, the file could be created by someone else in the meanwhile.
3. mkstemp is an alternative. However it is not C-standard and not really compatible with the way `os.tempname` works. What unpatched lua does (if LUA_USE_LINUX is defined) is to create a file and then close it (it does not even delete it).

Conclusion: it is a bug in the Lua API.

Reply:
@danpetry Yes, exactly. It causes different code to be built for native, which defeats the purpose of being able to test your app in your PC, and have that app be as close as possible to what will run on the MCU later.

Reply:
You could create another #define LUA_RIOT_NATIVE or something which only deals with the offending lines in loslib.c and leaves the rest?
It's my understanding that "no compiler warnings" is a golden rule, maybe Jose can clarify having been a maintainer for longer

Reply:
As far as I see, this problem vanishes if -DLUA_USE_LINUX is not present. Right? As @jcarrano pointed out in this thread, it might be a good idea to run these without the LUA_USE_LINUX flag. That should solve the issue.

Reply:
We could remove the warning, but the code would still be wrong. It is the very definition of [`os.tmpname`](https://www.lua.org/manual/5.3/manual.html#pdf-os.tmpname) that is wrong, and the user should instead use [`io.tmpfile`](https://www.lua.org/manual/5.3/manual.html#pdf-io.tmpfile) instead. Removing the function would be a better idea, we break the "official" lua api, but again, people should not be using that anyways.

Reply:
@jia200x No, it's exactly the opposite. The warning vanishes with -DLUA_USE_LINUX. But then, not only is the code different: the function does different things too!. See https://www.lua.org/manual/5.3/manual.html#pdf-os.tmpname . With DLUA_USE_LINUX it creates a file, which you must remove afterwards!!!!!

Reply:
@jcarrano Then I would vote for removing/patching the function from RIOT. There are other packages that already do this (e.g [LoRaMAC](https://github.com/RIOT-OS/RIOT/blob/master/pkg/semtech-loramac/patches/0001-adapt-for-RIOT.patch)).

If there's a not-so-desired behavior, it's OK to patch it.

Reply:
@jia200x @danpetry Done, os.tmpname is gone. I am accepting the fact that we will have to patch lua and there is no way around.

Reply:
yes, I think it's ok to do it. Although those modules are part of the Lua implementation, they are not part of the "Lua interpreter". There are high chances Lua for RIOT is not exactly the same as Lua for Linux. Use cases tend to be different IMO

Reply:
Cool, worth documenting somewhere? Worth starting maybe a wiki page on Lua for RIOT or something which can grow?

Reply:
I tend to prefer Doxygen pages (or similar), since the wiki is hard to maintain. See #8516

Reply:
> * remove os.tmpname using the same approach as used with os.rename and os.clock (see first patch)
> * record that these three functions aren't supported in doc.txt

Done

Reply:
Great, thanks.